src/api/c/cast.cpp

```diff
@@ -24,7 +24,7 @@ using namespace detail;
 
 static af_array cast(const af_array in, const af_dtype type)
 {
-    const ArrayInfo& info = getInfo(in);
+    const ArrayInfo& info = getInfo(in, false, true);
```

Comment:
I just merged from hotfix-3.4.2, which had that change. The github diff in the "files changed" tab actually doesn't show the diff like it is showing here. Must be github bug...